src/lib/core/components/ScatterplotRsquareTable.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  XYPlotDataSeries,
+  XYPlotData,
+} from '@veupathdb/components/lib/types/plots';
+import { VariableTreeNode } from '../types/study';
+
+interface facetDataProps {
+  label: string;
+  data?: XYPlotData | undefined;
+}
+
+export interface Props {
+  data: XYPlotDataSeries[] | facetDataProps[] | undefined;
+  isFaceted: boolean;
+  overlayVariable?: VariableTreeNode;
+  facetVariable?: VariableTreeNode;
+}
+
+export function ScatterplotRsquareTable({
+  data,
+  isFaceted,
+  overlayVariable,
+  facetVariable,
+}: Props) {
+  // non-facet or facet data
+  const filteredData = !isFaceted
+    ? overlayVariable != null
+      ? (data as XYPlotDataSeries[])?.filter((data) =>
+          data?.name?.includes(', Best fit')
+        )
+      : (data as XYPlotDataSeries[])?.filter((data) =>
+          data?.name?.includes('Best fit')
+        )
+    : (data as facetDataProps[]).filter((element) => element.data != null);
+
+  if (!isFaceted) {
+    return (
+      <div className={'ScatterRsquareTable'}>
+        <table>
+          <tbody>
+            <tr>
+              <th>
+                {overlayVariable != null ? overlayVariable.displayName : 'Name'}
+              </th>
+              <th className="numeric">
+                R<sup>2</sup> (Best fit)
+              </th>
+            </tr>
+            {filteredData != null
+              ? (filteredData as XYPlotDataSeries[]).map((data) => (
+                  <tr>
+                    <td>{data?.name?.split(', Best fit')[0]}</td>
+                    <td>{data.r2 ?? 'N/A'}</td>
```

Comment:
This line is giving me a TypeScript error - `Property 'r2' does not exists on type 'XYPlotDataSeries'`.

I'm guessing you'll need to introduce some new types in `ScatterplotVisualization.tsx`, perhaps something like this:

```
interface XYPlotDataSeriesWithRSquare extends XYPlotDataSeries {
  r2?: number;
}

interface XYPlotDataWithRSquare {
  series: XYPlotDataSeriesWithRSquare[];
}
```

You'd then have to update `XYPlotDataWithCoverage` to something like `XYPlotDataWithRSquareAndCoverage`:

```
// define XYPlotDataWithRSquareAndCoverage
interface XYPlotDataWithRSquareAndCoverage extends CoverageStatistics {
  dataSetProcess: XYPlotDataWithRSquare | FacetedData<XYPlotDataWithRSquare>;
  // change these types to be compatible with new axis range
  yMin: number | string | undefined;
  yMax: number | string | undefined;
}
```

Let me know if you have another idea!

Reply:
@jtlong3rd Thank you for your comment and suggestions! 👍 I am on vacation today but I found your nice explanation. Actually it is quite strange as I have defined it at XYPlotDataSeries as you approved the PR, [#274 at web-components](https://github.com/VEuPathDB/web-components/pull/274),  and updated the web-components version at this PR. Can you please try `git pull origin scatter-r2-table` and then `yarn install` ? I don't have type error.

I may have more time to check this tomorrow if you would still have the issue. Thanks in advance!

Reply:
@moontrip Crisis averted! I was `yarn link`ed to an old version of web-components that didn't have `r2` in `XYPlotData`. All I had to do was `yarn unlink`.